iotfunctions/util.py

```diff
@@ -958,26 +958,29 @@ def save(self):
         '''
         Write trace to COS
         '''
+        try:
+            save_to_file = self.parent.save_trace_to_file
+        except AttributeError:
+            save_to_file = self.save_trace_to_file
 
+        trace = None
         if len(self.data) == 0:
-            trace = None
             logger.debug('Trace is empty. Nothing to save.')
+        else:
+            trace = str(self.as_json())
+
+        if trace is not None and save_to_file:
```

Comment:
By this logic, it will not save the file to the local machine. But it will still push the file to cos.  Correct me if I'm wrong.

Reply:
it will attempt to save empty (None) traces to db I agree - and this might cause headaches
if trace is non-empty and the flag set it should save it locally - at least it did when I tried it.